server/routes/update-goal/UpdateGoalController.ts

```diff
@@ -0,0 +1,79 @@
+import { NextFunction, Request, Response } from 'express'
+import locale from './locale.json'
+import ReferentialDataService from '../../services/sentence-plan/referentialDataService'
+import URLs from '../URLs'
+import transformRequest from '../../middleware/transformMiddleware'
+import UpdateGoalPostModel from './models/UpdateGoalPostModel'
+import StepModel from '../shared-models/StepModel'
+import validateRequest from '../../middleware/validationMiddleware'
+import { NewStep } from '../../@types/StepType'
+
+export default class UpdateGoalController {
+  constructor(private readonly referentialDataService: ReferentialDataService) {}
+
+  private render = async (req: Request, res: Response, next: NextFunction) => {
+    const { uuid } = req.params
+    const { errors } = req
+
+    const sortedAreasOfNeed = this.referentialDataService.getSortedAreasOfNeed()
+    const goal = await req.services.goalService.getGoal(uuid)
+    const popData = req.services.sessionService.getSubjectDetails()
+    const mainAreaOfNeed = sortedAreasOfNeed.find(areaOfNeed => areaOfNeed.name === goal.areaOfNeed.name)
+    const relatedAreasOfNeed = goal.relatedAreasOfNeed.map(need => need.name)
+
+    return res.render('pages/update-goal', {
+      locale: locale.en,
+      data: {
+        goal,
+        popData,
+        mainAreaOfNeed,
+        relatedAreasOfNeed,
+      },
+      errors,
+    })
+  }
+
+  private saveAndRedirect = async (req: Request, res: Response, next: NextFunction) => {
+    const { uuid } = req.params
+    const { steps } = req.body
+    let error
+
+    const goal = await req.services.goalService.getGoal(uuid)
+
+    goal.steps.forEach((step: StepModel, index) => {
+      if (step.uuid !== steps[index].uuid) {
+        error = new Error('different steps were submitted')
+      }
+    })
```

Comment:
Something like 
```
if (goal.steps.some((step: StepModel, index) => step.uuid !== steps[index].uuid)) {
  return next(Error());
}
```
might make more sense here, though this will render a 500 error page. 

If you provide it a `HttpError` you can specify the status code, so for this case, probably a 400 bad request.

Reply:
Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/170/commits/f997e1e926db9079fb24439d13ebd03e98c3f420